### DISCUSSION TITLE
Is it possible to only query index with specific fields when using <Index /> component?

### DISCUSSION THREAD

**Discussion by Sartonon** [algolia/instantsearch#5348] [Q&A] [answered]
I would like to use <Index /> component to query suggestions index and only query one field (query field in suggestions index). I don't want to include all filters coming from other hooks in the tree.

You get the following error if you try to add range filter in the same tree:  `invalid numeric attribute(price), attribute not specified in numericAttributesForFiltering setting`

Example: https://codesandbox.io/s/elastic-davinci-5kzfij?file=/src/App.js

**Comment by Haroenv** [comment]
For this to work you need to change your index settings (on the dashboard) to add price to `attributesForFaceting` or `numericAttributesForFiltering`, depending on what it's set to now you may need to remove `filterOnly` as well

**Reply by Sartonon** [reply]
Problem is that `autocomplete_demo_products_query_suggestions` doesn't have a price field

**Reply by Haroenv** [reply]
That's indeed not easy to solve. There's a couple solutions possible:

- restructure indices to inherit parameters from root

This would be hard with your setup, as suggestions are within the layout

- remove the filter from indices that don't have those attributes

https://codesandbox.io/s/relaxed-david-ofo3hk?file=/src/App.js:214-645

```js
const searchClient = algoliasearch(
  "latency",
  "6be0576ff61c053d5f9a3225e2a90f76"
);
const search = searchClient.search;
searchClient.search = (queries) => {
  return search(
    queries.map(({ indexName, params }) => {
      if (indexName === "autocomplete_demo_products_query_suggestions") {
        delete params.numericFilters;
        delete params.facetFilters;
      }

      return { indexName, params };
    })
  );
};
```

- implement parentIndexId

This is not something we have yet, but we thought of implementing a way to bypass the regular "inheriting" of search parameters by allowing `parentIndexId={null}` on indices, but it still has some question marks on how it should be implemented.


- add the facet to the index, even if it has no effect

This means however that as soon as you filter, there will be no results

**Reply by Sartonon** [reply]
> * restructure indices to inherit parameters from root
> 
> This would be hard with your setup, as suggestions are within the layout

Not sure what you mean by this 🤔 

> * remove the filter from indices that don't have those attributes
> 
> https://codesandbox.io/s/relaxed-david-ofo3hk?file=/src/App.js:214-645

This could work for now. I'll try this approach for now :)

**Reply by Haroenv** [reply]
If you would have a layout like this

```
<InstantSearch>
  <Index suggestions>
  <Index hits>
</InstantSearch>
```

the filters wouldn't be inherited

**Reply by Sartonon** [reply]
Ok good to know!

**Reply by Sartonon** [reply]
> * remove the filter from indices that don't have those attributes
> 
> https://codesandbox.io/s/relaxed-david-ofo3hk?file=/src/App.js:214-645
> 
> ```js
> const searchClient = algoliasearch(
>   "latency",
>   "6be0576ff61c053d5f9a3225e2a90f76"
> );
> const search = searchClient.search;
> searchClient.search = (queries) => {
>   return search(
>     queries.map(({ indexName, params }) => {
>       if (indexName === "autocomplete_demo_products_query_suggestions") {
>         delete params.numericFilters;
>         delete params.facetFilters;
>       }
> 
>       return { indexName, params };
>     })
>   );
> };
> ```

This worked nicely for our project for now :)

**Comment by Sartonon** [comment, marked as answer]
> That's indeed not easy to solve. There's a couple solutions possible:
> 
> * restructure indices to inherit parameters from root
> 
> This would be hard with your setup, as suggestions are within the layout
> 
> * remove the filter from indices that don't have those attributes
> 
> https://codesandbox.io/s/relaxed-david-ofo3hk?file=/src/App.js:214-645
> 
> ```js
> const searchClient = algoliasearch(
>   "latency",
>   "6be0576ff61c053d5f9a3225e2a90f76"
> );
> const search = searchClient.search;
> searchClient.search = (queries) => {
>   return search(
>     queries.map(({ indexName, params }) => {
>       if (indexName === "autocomplete_demo_products_query_suggestions") {
>         delete params.numericFilters;
>         delete params.facetFilters;
>       }
> 
>       return { indexName, params };
>     })
>   );
> };
> ```
> 
> * implement parentIndexId
> 
> This is not something we have yet, but we thought of implementing a way to bypass the regular "inheriting" of search parameters by allowing `parentIndexId={null}` on indices, but it still has some question marks on how it should be implemented.
> 
> * add the facet to the index, even if it has no effect
> 
> This means however that as soon as you filter, there will be no results
